src/pages/AddPage/AddPage.jsx

```diff
@@ -1,7 +1,46 @@
 import React from "react";
+import "./AddPage.scss";
+import { useState } from "react";
 
 const AddPage = () => {
-  return <h2>add</h2>;
+  const url = "http://localhost:3030";
+  const [text, setText] = useState("");
+  const [error, setError] = useState(null);
+  const [done, setDone] = useState(null);
+  const handleAddTodo = async () => {
+    await postTodo();
+  };
+
+  const postTodo = async () => {
+     try {
+      await fetch(`${url}/todos`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ text }),
+    });
+    setDone("done");
+  } catch (error) {
+    setError(error);
```

Comment:
In this case, it's better if you pass the error text to the function instead of passing the error object. Something like: `setError(e.message)`. Then in the jsx you can just use ``error` (that is a string)`.

Reply:
Done.